gm4_mountaineering/data/gm4_mountaineering/functions/main.mcfunction

```diff
@@ -0,0 +1,13 @@
+# Run ski-related functions
+execute as @a[gamemode=!spectator] at @s run function gm4_mountaineering:ski_effects/skis_equipped
+
+# Adds jump boost effects when holding poles
+execute as @a[gamemode=!spectator,predicate=gm4_mountaineering:holding_poles] at @s run function gm4_mountaineering:ski_effects/poles_jump_boost
+
+# Adds vertical climbing abilities when wearing crampons and not holding anything
+execute as @a[gamemode=!spectator,predicate=gm4_mountaineering:wearing_crampons] at @s run function gm4_mountaineering:climbing_effects/crampons_equipped
+
+# Clear any crampon effects when crampons are removed
+execute as @a[gamemode=!spectator,predicate=!gm4_mountaineering:wearing_crampons] at @s run function gm4_mountaineering:climbing_effects/clear_crampon_effects
```

Comment:
It would be better if you used a `player_submain` function instead, so you can reduce the amount of `@a at @s` you have to use.